src/main/scala/io/scalac/tezos/translator/routes/LibraryRoutes.scala

```diff
@@ -23,109 +28,169 @@ class LibraryRoutes(
   captchaConfig: CaptchaConfig,
   adminEmail: EmailAddress
 )(implicit as: ActorSystem, ec: ExecutionContext) extends HttpRoutes {
-  import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
   import io.circe.generic.auto._
 
-  override def routes: Route =
-    (path("library") & pathEndOrSingleSlash) {
-      (post & withReCaptchaVerify(log, captchaConfig)(as) & withLibraryDTOValidation) { libraryDto =>
-        val sendEmailF = libraryDto.email match {
-          case Some(_) =>
-            val e = SendEmail.approvalRequest(libraryDto, adminEmail)
+  private val libraryCaptchaEndpoint: Endpoint[Option[String], (DTO.ErrorDTO, StatusCode), Unit, Nothing] = Endpoints.captchaEndpoint(captchaConfig).in("library")
+  private  val libraryEndpoint: Endpoint[Unit, Unit, Unit, Nothing] = Endpoints.baseEndpoint.in("library")
+
+  private val libraryAddEndpoint: Endpoint[(Option[String], LibraryEntryRoutesDto), (DTO.ErrorDTO, StatusCode), StatusCode, Nothing] =
+    libraryCaptchaEndpoint
+      .in(jsonBody[LibraryEntryRoutesDto])
+      .post
+      .out(statusCode)
+      .description("Will add LibraryEntryRoutesDto to storage")
+
+  private val getDtoEndpoint: Endpoint[(Option[String], Option[Int], Option[Int]), ErrorResponse, Seq[LibraryEntryDTO], Nothing] =
+    libraryEndpoint
+      .in(maybeAuthHeader)
+      .in(offsetQuery.and(limitQuery))
+      .errorOut(errorResponse)
+      .out(jsonBody[Seq[LibraryEntryDTO]])
+      .get
+      .description("Will return sequence of LibraryEntryRoutesDto or LibraryEntryRoutesAdminDto if auth header provided")
+
+  private val putEntryEndpoint: Endpoint[(String, String, String), ErrorResponse, StatusCode, Nothing] =
+    libraryEndpoint
+      .in(auth.bearer)
+      .in(uidQuery.and(statusQuery))
+      .errorOut(errorResponse)
+      .out(statusCode)
+      .put
+      .description("Will change the status of entry with passed uid")
+
+  private val deleteEntryEndpoint: Endpoint[(String, String), ErrorResponse, StatusCode, Nothing] =
+    libraryEndpoint
+      .in(auth.bearer)
+      .in(uidQuery)
+      .errorOut(errorResponse)
+      .out(statusCode)
+      .delete
+      .description("Will delete an entry with passed uid")
+
+  private def addNewEntryRoute(): Route =
+    libraryAddEndpoint.toRoute {
+      (withReCaptchaVerify(_, log, captchaConfig))
+        .andThenFirstE((validateLibraryEntryRoutesDTO _).tupled)
```

Comment:
WDYT about doing it in this way ?
```
    libraryAddEndpoint.toRoute {
      (withReCaptchaVerify(_, log, captchaConfig))
        .andThenFirstE { t: (Unit, LibraryEntryRoutesDto) => DTOValidation.validateDto(t._2) }
        .andThenFirstE(addNewEntry)
    }
```
then no need of auxiliary function `validateLibraryEntryRoutesDTO ` will be needed.